lib/request.js

```diff
@@ -159,6 +159,11 @@ Object.defineProperties(Request.prototype, {
       return this.socket.encrypted ? 'https' : 'http'
     }
   },
+  referrer: {
+    get () {
+      return this.raw.headers.referrer || this.raw.headers.referer || ''
```

Comment:
why does it support one `r`? Maybe add a comment to it.
I would also cache the headers object: 

```suggestion
      const headers = this.raw.headers
      return headers.referrer || headers.referer || ''
```

Reply:
one `r` is supported because `Referer` is what is sent by browsers (it's a historical misspelling).  See my [comment](https://github.com/fastify/fastify/pull/2953#discussion_r600956245) for more explanation